soc/projects/compiler.md

```diff
@@ -7,44 +7,49 @@ title:  Compiler Projects – Summer of Code
 
 {% include toc.html %}
 
-## C Linter
+## Improved code-generation for un-inferred code
 
-Memory errors in Julia's underlying C code are sometimes difficult to trace, and missing garbage-collector "roots" (GC roots) can lead to segfaults and other problems. One potential way to make it easier to find such errors would be to write a package that checks Julia's `src/` directory for missing GC roots. A Julia-based solution might leverage the [Clang.jl](https://github.com/ihnorton/Clang.jl) package to parse the C code, determine which call chains can trigger garbage collection, and then look for objects that lack GC root protection. Alternatively, the same strategy might be implemented in C++ by writing a plugin for [Clang's static analyzer](http://clang-analyzer.llvm.org/). Another attractive approach is to leverage [coccinelle](http://coccinelle.lip6.fr/).
+Currently the quality (perfomance) of generated code does not degrade particularly gracefully as the quality of type-inference degrades. There are a number of difference projects possible here. Some example optimization opportunities include:
 
-**Expected Results**: A tool that, when run against Julia's `src/` directory, highlights lines that need to have additional GC roots added.
+- Calling convention for varargs functions (https://github.com/JuliaLang/julia/issues/5402)
+- Call-site splatting optimizations (https://github.com/JuliaLang/julia/issues/13359)
+- Keyword argument inference / inlining
+- More calling conventions, permitting improved performance (unboxed argument passing) when the direct method target is not known
+- Value domination and liveness analysis for optimizing stack space and gc-root allocation
 
-**Required Skills**: Familiarity with C, runtimes, and experience with (or interest in) static analysis.
+**Recommended Skills**: Experience with reading LLVM IR or assembly. An understanding of formal logic and using it develop improved code transformations.
```

Comment:
`using it develop` -> `using it to develop` or perhaps `and its application to improving code transformations`?